### PR TITLE
Makes CodeListEntries translatable

### PR DESCRIPTION
**Pull request by cschierle**
CodeListEntry now implements the Translatable interface.
Class usages in Controllers and other classes have been updated to include the additional required generic.

Translation capabilities are not actively used yet, but will be after upcoming UI changes

Fixes: SIRI-239

### CODE REVIEW

File: src/main/java/sirius/biz/codelists/CodeListEntry.java [hunk 1]
```diff
@@ -24,7 +26,8 @@
  */
 @SuppressWarnings("squid:S1214")
 @Explain("We rather keep the constants here, as this emulates the behaviour and layout of a real entity.")
-public interface CodeListEntry<I, L extends BaseEntity<I> & CodeList> extends Entity {
+public interface CodeListEntry<I, L extends BaseEntity<I> & CodeList, T extends BasicTranslations<?>>
```
Comment (bwiedmann) [inline src/main/java/sirius/biz/codelists/CodeListEntry.java:29]: 🥇 👍

File: src/main/java/sirius/biz/codelists/CodeListController.java [hunk 2]
```diff
@@ -31,7 +32,7 @@
  * @param <L> the effective entity type used to represent code lists
  * @param <E> the effective entity type used to represent code list entries
  */
-public abstract class CodeListController<I, L extends BaseEntity<I> & CodeList, E extends BaseEntity<I> & CodeListEntry<I, L>>
+public abstract class CodeListController<I, L extends BaseEntity<I> & CodeList, T extends BasicTranslations<?>, E extends BaseEntity<I> & CodeListEntry<I, L, T>>
```
Comment (idlira) [inline src/main/java/sirius/biz/codelists/CodeListController.java:35]: Update JavaDoc to describe `T`

File: src/main/java/sirius/biz/codelists/CodeLists.java [hunk 3]
```diff
@@ -51,7 +52,7 @@
  * @param <L> the effective entity type used to represent code lists
  * @param <E> the effective entity type used to represent code list entries
  */
-public abstract class CodeLists<I, L extends BaseEntity<I> & CodeList, E extends BaseEntity<I> & CodeListEntry<I, L>> {
+public abstract class CodeLists<I, L extends BaseEntity<I> & CodeList, T extends BasicTranslations<?>, E extends BaseEntity<I> & CodeListEntry<I, L, T>> {
```
Comment (idlira) [inline src/main/java/sirius/biz/codelists/CodeLists.java:55]: Document `T` in JavaDoc

**Review by ymo-sci — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by bwiedmann — APPROVED** [review]

File: src/main/java/sirius/biz/codelists/CodeLists.java [hunk 4]
```diff
@@ -49,9 +50,10 @@
  *
  * @param <I> the type of database IDs used by the concrete implementation
  * @param <L> the effective entity type used to represent code lists
+ * @param <T> the effective entity type used to represent translations
  * @param <E> the effective entity type used to represent code list entries
  */
-public abstract class CodeLists<I, L extends BaseEntity<I> & CodeList, E extends BaseEntity<I> & CodeListEntry<I, L>> {
+public abstract class CodeLists<I, L extends BaseEntity<I> & CodeList, T extends BasicTranslations<?>, E extends BaseEntity<I> & CodeListEntry<I, L, T>> {
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/CodeLists.java:56]: do we really need to expose this in the type-system or can't we rely on the interface here?
-> there are tons of references to this type...

Comment (andyHa) [reply src/main/java/sirius/biz/codelists/CodeLists.java:56]: Also shouldn't there be a getter for translated values anywhere?

Comment (cschierle) [reply src/main/java/sirius/biz/codelists/CodeLists.java]: true, the exposed type is not really necessary and has been removed.

added getter for translated values as well.

**Review by bwiedmann — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Pull request merged**